File: tests/test_ComputeSeries.cxx
/**
*  @file    test_ComputeSeries.cxx
*  @brief   Tests for the class ComputeSeries.
*  @author  Francois Roy
*  @date    2/20/2016
*  @version 1.0.0
*/

#include "gtest/gtest.h"
#include "ComputeSeries.h"

/**
 *  \test Test publicMemberFunction
 */
// Create a dummy class to inherit the abstract class ComputeSeries
class Dummy :public ComputeSeries {
public:
    Dummy();
    ~Dummy();
private:
    double fct(int n);
};
Dummy::Dummy(){}
Dummy::~Dummy(){}
double Dummy::fct(int n) {
    double var =  (double) n;
    // Geometric series with r = 1/2
    double expression = pow(1/2.0,var);
    return expression;
}

TEST(ComputeSeriesTest, testSumForward) {
    Dummy d;
    double out = d.getSumForward(1e-20);
    // Geometric series exact solution is out = 1/(1-r), where r = 1/2
    EXPECT_NEAR(out, 2.0, 1e-20);
}

TEST(ComputeSeriesTest, testSumKahan) {
    Dummy d;
    double out = d.getSumKahan(1e-20);
    // Geometric series exact solution is out = 1/(1-r), where r = 1/2
    EXPECT_NEAR(out, 2.0, 1e-20);
}

File: tests/test_ComputeSeries.py
import unittest
import numpy as np
from heat.wrapper import ComputeSeries


class Dummy(ComputeSeries):
    """Dummy class to implement the Geometric series.
    """

    def fct(self, n):
        """Define the function to be summed.
        """
        # Geometric series with r = 1/2
        expression = pow(1/2.0,n)
        return expression

class TestComputeSeries(unittest.TestCase):
    """Test the wrapper class ComputeSeries.

    """
    def test_getSumForward(self):
        """Test sumForward.
        """
        d = Dummy()
        self.assertEqual(0.125, d.fct(3))
        #d.getSumForward(1.0e-9,8)